src/forms/PayoutDetailsForm/PayoutDetailsFormCompany.js

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import { bool, object, string } from 'prop-types';
+import { compose } from 'redux';
+import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
+import { FieldTextInput, FieldArray, IconClose, InlineTextButton } from '../../components';
+import * as validators from '../../util/validators';
+
+import PayoutDetailsAddress from './PayoutDetailsAddress';
+import PayoutDetailsBankDetails from './PayoutDetailsBankDetails';
+import PayoutDetailsPersonalDetails from './PayoutDetailsPersonalDetails';
+import { stripeCountryConfigs } from './PayoutDetailsForm';
+import css from './PayoutDetailsForm.css';
+
+const PayoutDetailsFormCompanyComponent = ({ fieldRenderProps }) => {
+  const {
+    id,
+    disabled,
+    form,
+    intl,
+    values,
+    form: {
+      mutators: { push },
+    },
+  } = fieldRenderProps;
+  const { country } = values;
+
+  const companyNameLabel = intl.formatMessage({ id: 'PayoutDetailsForm.companyNameLabel' });
+  const companyNamePlaceholder = intl.formatMessage({
+    id: 'PayoutDetailsForm.companyNamePlaceholder',
+  });
+  const companyNameRequired = validators.required(
+    intl.formatMessage({
+      id: 'PayoutDetailsForm.companyNameRequired',
+    })
+  );
+
+  const companyTaxIdLabel = intl.formatMessage({
+    id: `PayoutDetailsForm.companyTaxIdLabel.${country}`,
+  });
+  const companyTaxIdPlaceholder = intl.formatMessage(
+    {
+      id: 'PayoutDetailsForm.companyTaxIdPlaceholder',
+    },
+    {
+      idName: companyTaxIdLabel,
+    }
+  );
+  const companyTaxIdRequired = validators.required(
+    intl.formatMessage(
+      {
+        id: 'PayoutDetailsForm.companyTaxIdRequired',
+      },
+      {
+        idName: companyTaxIdLabel,
+      }
+    )
+  );
+
+  const showPersonalAddressField =
+    country &&
+    stripeCountryConfigs(country).companyConfig &&
+    stripeCountryConfigs(country).companyConfig.personalAddress;
+
+  const showAdditionalOwnersField =
+    country &&
+    stripeCountryConfigs(country).companyConfig &&
+    stripeCountryConfigs(country).companyConfig.additionalOwners;
+
+  const hasAdditionalOwners = values.company && values.company.additionalOwners;
+  const hasMaxNumberOfAdditionalOwners =
+    hasAdditionalOwners && values.company.additionalOwners.length >= 4;
```

Comment:
We don't want to use magic numbers in the code. So, I'd suggest that this number for is either 
1) coming from config: `maxNumberOfAdditionalOwners` (might be unnecessary if the value is just 4 or nothing) or
2) it is a constant defined at the beginning of this file:
```
// In EU, there can be a maximum of 4 additional owners
const MAX_NUMBER_OF_ADDITIONAL_OWNERS = 4;
```